processing/src/main/java/org/apache/druid/query/aggregation/AggregatorFactory.java

```diff
@@ -219,7 +220,13 @@ public AggregatorFactory getMergingFactory(AggregatorFactory other) throws Aggre
    *
    * Refer to the {@link ColumnType} javadocs for details on the implications of choosing a type.
    */
-  public abstract ColumnType getType();
+  public ColumnType getColumnType()
+  {
+    if (getType() == ValueType.COMPLEX) {
+      return ColumnType.ofComplex(getComplexTypeName());
```

Comment:
getComplexTypeName might not be implemented for aggregators that aren't ingestible (i.e. no ComplexMetricSerde), so this should catch IllegalStateException and treat it as an unknown complex type.